### DISCUSSION TITLE
'CompoundSelect' object has no attribute 'mapper' - bug or mistake on my part?

### DISCUSSION THREAD

**Discussion by andrewcooke** [sqlalchemy/sqlalchemy#6271] [Usage Questions] [answered]
i am trying to call aliased(union(...)) that represents

```
SELECT source.id AS source_id 
FROM source JOIN (statistic_journal JOIN statistic_journal_integer ON statistic_journal.id = statistic_journal_integer.id) ON source.id = statistic_journal.source_id JOIN statistic_name ON statistic_name.id = statistic_journal.statistic_name_id 
WHERE statistic_name.name LIKE :name_1 AND statistic_journal_integer.value > :value_1 UNION SELECT source.id AS source_id 
FROM source JOIN (statistic_journal JOIN statistic_journal_float ON statistic_journal.id = statistic_journal_float.id) ON source.id = statistic_journal.source_id JOIN statistic_name ON statistic_name.id = statistic_journal.statistic_name_id 
WHERE statistic_name.name LIKE :name_2 AND statistic_journal_float.value > :value_2
```

and am getting the error 
`AttributeError: 'CompoundSelect' object has no attribute 'mapper'`

afaict this used to work (although i haven't triangulated when it started to fail).  am i doing something wrong or should i raise a bug report?  (i'm asking first since it will take some effort to make a reproducible example).

thanks,
andrew

**Comment by zzzeek** [comment, marked as answer]
there's probably a bug if you're using Query, sure, and i would want to see the full stack trace.

the difficulty in doing ORM unions is one of the main reasons I wanted to get rid of Query though, if you work with select() directly it is likely easier to get the exact SQL you want.

**Reply by andrewcooke** [reply]
Yeah it's session.query(...) https://github.com/sqlalchemy/sqlalchemy/issues/6274

I'll look at rewriting my code to use select.
